### PR TITLE
tests: simplify the pr-title check

### PR DESCRIPTION
**Pull request by sergiocazzolato**
The check-pr-title.py was created when we used travis (a long time ago).

Now, in github actions we can access to the pr title from ${{ github.event.pull_request.title }}

The idea of this change is to avoid checking in the github api and simplify the check-pr-title.py tool.

This also is required to be able to use this check in private projects where a unauthenticated api call is not allowed.

Also, I am removing from the actions workflow the pr number variables which are not used anymore.

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -102,8 +102,7 @@ jobs:
       # version of go.
       PATH: /snap/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:${{ github.workspace }}/bin
       GOROOT: ""
-      GITHUB_PROJECT_URL: ${{ github.server_url }}/${{ github.repository }}
-      GITHUB_PULL_REQUEST: ${{ github.event.number }}
+      GITHUB_PULL_REQUEST_TITLE: ${{ github.event.pull_request.title }}
```
Comment (ernestl) [inline .github/workflows/test.yaml:105]: run-checks still references GITHUB_PULL_REQUEST. The condition of running PR title check needs to be updated.

**Review by ernestl — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yaml (hunk 1 above)
Comment (ernestl) [reply .github/workflows/test.yaml:105]: I added a commit to fix this: https://github.com/snapcore/snapd/pull/14021/commits/d6a29bfada85f0148da817f114c8d3880cb4f8fb

**Review by ZeyadYasser — APPROVED** [review]

**Review by ernestl — APPROVED** [review]
Approve, but last commit is my own so someone else needs to look at that :)

**Review by Meulengracht — APPROVED** [review]
Thank you

**Pull request merged**